pkg/reporting/reporter.go

```diff
@@ -5,33 +5,36 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/openshift/must-gather-clean/pkg/omitter"
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
 	"gopkg.in/yaml.v3"
 	"k8s.io/klog/v2"
 )
 
 type Report struct {
-	Replacements []map[string]string `yaml:"replacements,omitempty"`
-	Omissions    []string            `yaml:"omissions,omitempty"`
+	Replacements []obfuscator.Replacement `yaml:"replacements,omitempty"`
+	Omissions    []string                 `yaml:"omissions,omitempty"`
 }
 
 type Reporter interface {
 	// WriteReport writes the final report into the given path, will create folders if necessary.
 	WriteReport(path string) error
 
-	// CollectOmitterReport will call the Report method on the omitter and collect its omissions.
-	CollectOmitterReport(omitter omitter.ReportingOmitter)
+	// CollectOmitterReport collects the omitter's omission results.
+	CollectOmitterReport(omitter []string)
 
 	// CollectObfuscatorReport will call the Report method on the obfuscator and collect the individual obfuscation results.
-	CollectObfuscatorReport(obfuscatorReport []map[string]string)
+	CollectObfuscatorReport(obfuscatorReport []obfuscator.ReplacementReport)
 }
 
 type SimpleReporter struct {
-	replacements []map[string]string
+	replacements []obfuscator.Replacement
 	omissions    []string
 }
 
+var _ Reporter = (*SimpleReporter)(nil)
```

Comment:
what does that do? should we typedef this thing?

Reply:
Compile time check member functions of SimpleReporter implement functions of interface Reporter

Reply:
mind blown, why are we not returning the Reporter interface in the New func below?

Reply:
Good point. I'll change that.